src/pagination.rs

```diff
@@ -279,34 +286,19 @@ fn interpret_permalink(
     } else {
         let mut attributes = document::permalink_attributes(&doc.front, &doc.file_path);
         let pagination_attr = pagination_attributes(index_num as i32, config.include);
-        pagination_attr.into_iter().for_each(|(k, v)| {
-            attributes.insert(k, v);
-        });
-
+        attributes.extend(pagination_attr.into_iter());
         permalink::explode_permalink(&config.permalink, &attributes)?
     })
 }
 
-fn range_for_page(per_page: i32, nb_posts: usize) -> Range<usize> {
-    let nb_posts = nb_posts as i32;
+fn range_for_page(index: usize, per_page: i32, nb_posts: usize) -> Range<usize> {
+    let start = (index - 1) * per_page as usize;
+    let end = start + per_page as usize;
     // make sure `end` is not beyond capacity
-    0..if per_page < nb_posts {
-        per_page as usize
+    trace!("start = {}, end = {}, nb_posts = {}", start, end, nb_posts);
+    start..if end < nb_posts {
```

Comment:
could we make this a `let end = ` instead?  It feels weird having the end part of the range be an entire conditional expression